fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+            special_tokens = set(init_vocab.keys())
+            special_inds = list(init_vocab.values())
+            if starting_index is None:
+                starting_index = max(special_inds) + 1
+        else:
+            special_tokens = set()
+            init_vocab = {}
+            if starting_index is None:
+                starting_index = 0
+
+        # At this point our vocab contains all the special tokens with their IDs, and we know from which ID to start the regular token mappings
+
+        regular_tokens = set(vocab.keys()) - special_tokens
+        regular_vocab = {
+            r_t: vocab[r_t] for r_t in regular_tokens
+        }  # These are only regular tokens with their original indices
+        regular_sorted = sorted(
+            regular_vocab.items(), key=lambda x: x[1], reverse=False
+        )  # regular tokens sorted by their ID in ascending order.
+
+        regular_vocab = {t[0]: i + starting_index for i, t in enumerate(regular_sorted)}
+        init_vocab.update(regular_vocab)
+        starting_index_new = max(regular_vocab.values()) + 1
+        return init_vocab, starting_index_new
+
+    @staticmethod
+    def build_special_token_list(special_tokens: Union[List, Set]) -> List:
+        """Creates a list of special token structures with consecutive indices, according to the following template
+            special token template:
+            {
+                'id': 0,
+                'content': '<UNK>',
+                'single_word': False,
+                'lstrip': False,
+                'rstrip': False,
+                'normalized': False,
+                'special': True
+            }
+
+
+        Args:
+            special_tokens (Union[List, Set]): _description_
+
+        Returns:
+            List: _description_
+        """
+        special_tokens = [
+            {
+                "id": i,
+                "content": v,
+                "single_word": False,
+                "lstrip": False,
+                "rstrip": False,
+                "normalized": False,
+                "special": True,
+            }
+            for i, v in enumerate(special_tokens)
+        ]
+        return special_tokens
+
+    @staticmethod
+    def get_special_tokens(tokenizer_json_inst: Dict, force_special: Optional[bool] = False) -> List:
+        """returns the special tokens from tokenizer defined by json_inst.
+            Note: An alternative would be to call tokenizer_inst.get_vocab(with_added_tokens), using with_added_tokens False and True, which
+            should've given us just regular and regular+special tokens, but for some reason both these options return the same output,
+            so we must resort to json parsing.
+
+
+        Args:
+            json_inst (Dict): _description_
+            force_special (Optional[bool], optional): If False, treats all added tokens as special. If True, considers only those that have "special": True as special. Defaults to False.
```

Comment:
According to the name, it sounds like True should force all to be special.

Reply:
I see your point. I changed the name to "enforce_special", so that if True, it enforces the "special" label